File: oop/python/lista01/Ex05.py

from math import ceil

from KeyboardInput import KeyboardInput

class Ex05:
    def getLesserDivisor(self, n) -> int:
        divisor = 2
        while n % divisor != 0 and divisor < ceil(n**(1/2)): divisor += 1
        return divisor if n % divisor == 0 else None

if __name__ == "__main__":
    lesserDivisorCalculator = Ex05()
    keyboardInput = KeyboardInput()

    print("Provide the base number: ")
    input = keyboardInput.readPositiveInt()

    divisor = lesserDivisorCalculator.getLesserDivisor(input)
    if (divisor == None): print("{0} is a prime number.".format(input))
    else: print("{0} is not a prime and it's lesser divisor is {1}.".format(input, divisor))

File: objetos/lista01_java/Ex05/Ex05.java
package Lista01.Ex05;

import Lista01.KeyboardInput;

public class Ex05 {
    public Integer getLesserDivisor(Integer n) {
        Integer divisor = 1;
        while(n % ++divisor != 0 && divisor < ((int) Math.sqrt(n)));
        return n % divisor == 0 ? divisor : null;
    }

    public static void main(String[] args) {
        Ex05 lesserDivisorCalculator = new Ex05();
        KeyboardInput keyboardInput = new KeyboardInput(System.in);

        System.out.println("Provide the base number: ");
        Integer input = keyboardInput.readPositiveInteger();

        Integer divisor = lesserDivisorCalculator.getLesserDivisor(input);
        if (divisor == null) System.out.println(input + " is a prime number.");
        else System.out.printf("%d is not a prime and it's lesser divisor is %d.%n", input, divisor);
    }
}
